src/main/pages/che-6/user-guide/commands-ide-macro.adoc

```diff
@@ -8,201 +8,223 @@ redirect_from: commands-ide-macro.html
-== Macros Auto-Completion
+[id="auto-completing-macros"]
+== Auto completing macros
 
-You can get auto-complete for all macros used in the editor. To activate this feature hit `<Ctrl+Space>` this will bring up a menu listing all the possible macros based on what’s been typed.
+You can auto-complete all macros used in the editor. To activate the auto-complete feature, press `Ctrl+Space`. A menu listing all the possible macros based on what is typed displays.
 
 image::commands/command-macros-autocompletion.png[]
 
-[id="use-commands"]
-== Use Commands
+[id="using-commands"]
+== Using commands
 
-You can use commands from multiple widgets:
+You can use commands from the following widgets:
 
-* Command palette +
-* Command toolbar +
-* Contextual menu in project explorer
+* *Command Palette*
 
-[id="command-palette"]
-== Command Palette
+* Command toolbar
 
-Since commands are often run in the heat of coding, you can use a hotkey to open the command palette.
+* Contextual menu in the *Project Explorer* view
 
-image::commands/command-palette.png[]
+[id="using-the-command-palette"]
+=== Using the Command Palette
 
-The command palette allows to quickly select a command to be executed. To call the command palette from the keyboard hit `<shift+F10>` and then use the cursor keys to navigate and enter to execute the command.
+Use the `Shift+F10` keys to open the *Command Palette*. Use the arrow keys to navigate through the list of commands. Use Enter to execute a selected command.  
+
+image::commands/command-palette.png[]
 
 [id="command-toolbar"]
-== Command Toolbar
+=== Using the command toolbar
 
-The command toolbar provides a way to execute the most common `Run` and `Debug` goals. It also provides access to all the executed commands and previews from a single place.
+Uisng the command toolbar, you can execute the most common `Run` and `Debug` goals. The toolbar provides access to all the executed commands and previews.
 
 image::commands/command-toolbar.png[]
 
-*Run and Debug Buttons*
-
-If you have commands defined for those goals, you can trigger them directly from those buttons.
+==== Uisng the Run and Debug buttons
 
-If you have multiple commands defined for the `Run` goal and if it’s the first time you are using the `Run` button, you’ll be asked to choose the default command associated with the button. The next click on the button will trigger the previously selected command.
+To trigger commands directly from the *Run* and the *Debug* buttons, you must define the commands for the `Run` and `Debug` goals.
 
-By doing a long click on the button you can select the command from the `Run` goal to execute. This command will become the default command associated with the `Run` button.
+When you use a button for the first time and if this button has multiple commands defined for a goal, you are asked to choose the default command associated with this button. The consecutive clicks of the button trigger the previously selected command.
```

Comment:
This is awkwardly worded. The opening sentence may flow better if we reverse the flow of it, something like this:

"You are asked to choose the default command associated with a button when you use this button for the first time and this button has multiple commands defined for a goal. Consecutive clicks of this button trigger the previously selected command."

Reply:
Nice suggestion! Done.